src/api/WebApi/WebApi/Controllers/WildBoarIotDataController.cs

```diff
@@ -15,9 +19,39 @@ public WildBoarIotDataController(IWildBoarIotDataService _wildBoarIotDataService
             wildBoarIotDataService = _wildBoarIotDataService;
         }
 
-        [HttpGet()]
-        public async Task<List<WildBoarIotData>> Get() =>
-            await wildBoarIotDataService.GetAsync();
+        [HttpGet("{sort?}/{format?}/{type?}/{weight?}/{occupied?}/{date_start?}/{date_end?}")]
+        public async Task<List<WildBoarIotData>> Get(string? sort, string? format, string? type, int weight,
+            bool occupied, DateTime? date_start, DateTime? date_end)
+        {
+            var getData = await wildBoarIotDataService.GetAsync();
+            
+            if (sort != null)
+            {
+                getData = getData.OrderBy(x => "x." + sort).ToList();
+            }
+            if (type != null)
+            {
+                getData = getData.FindAll(x => x.type == type);
+            } 
+            if (weight != null) {
+                getData = getData.FindAll(x => x.weight == weight);
+            } 
+            if (occupied != null) {
```

Comment:
Default value of uninitialized Bool is False not null. Therefore, this statement is always true.